### ISSUE TITLE
DPL-nnn Ensure frequent housekeeping of is_current column in lighthouse_samples

### ISSUE THREAD

**Issue by sdjmchattie** [sanger/crawler#422]
**Describe the Housekeeping**
Crawler is now responsible for ensuring that new samples entered into the `lighthouse_sample` table have the `is_current` column correctly populated.  This column should contain a 1 when the row represents the most recently received data for an RNA ID, while it should contain 0 if the row has been superseded by a more recently received piece of data.  While Crawler has been set up to try to update these values at every insert, there is a possibility of drift between reality and what this column shows.  This could be caused by manual edits to the data or a loss of connection during an update, for example.

The advice from Simon E. on the DBA team is to have a scheduled task run once per week or at some other interval which will parse the whole table and make sure all the values are currently set to what they should be.

**Additional context**
In an email from Simon:

>  I’d also advise a once a week query to check that all of the data is upto date, incase there’s been a processing error somewhere.
> It’s easy to get drift that can add up over time, if you’re only updating specific entries when they’re changed.
> (maybe someone inserts or deletes a couple of rows manually, because of some issue.  Or one of your processing runs crashes out or something, etc etc).
> I can write something for that, if it’d help.

I was thinking we could use something similar to the process used in the migration of the database for these two columns https://github.com/sanger/unified_warehouse/blob/develop/db/migrate/20210819162539_add_current_rna_id_to_lighthouse_sample.rb

In other words, we should run a process that sets all the rows back to `is_current = 0` and then execute the following:

```
UPDATE lighthouse_sample SET is_current = true
      WHERE id IN (
        SELECT most_recent.max_id FROM (
          SELECT rna_id, max(id) AS max_id
            FROM lighthouse_sample
            GROUP BY rna_id
        ) AS most_recent
```

This would probably run for about 30 minutes, but this would need to be tested and scheduled at the time of least impact.  There may be more optimal approaches -- this is just a suggestion.

**Comment by stevieing** [comment]
Delegated to GSU as a result of the Heron OKR
